### PR TITLE
Clickon refactor[DNM][DNTM]

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Click on is now a signal handler called by COMSIG_MOB_MOUSEDOWN

Currently break ai. Barely any testing done. Will break things. Not ready, i'm just putting this to have opinions on this concept

## Why It's Good For The Game

Currently, attacking, interacting etc requires you to click on and then release the mouse.
This will make the interaction proc as you are clicking immediatly
That should makes click smoother, so slashing, PBing, etc will be more consistent

## Changelog
:cl:
code: Clicking will be more reactive, as you just have to mousedown for the action to be done, rather than mousedown then mouseup
/:cl:

### CODE REVIEW

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -19,7 +19,6 @@
 /atom/Click(location, control, params)
 	if(flags_atom & INITIALIZED)
 		SEND_SIGNAL(src, COMSIG_CLICK, location, control, params, usr)
-		usr.ClickOn(src, location, params)
```
Comment (BraveMole) [inline code/_onclick/click.dm:22]: break ai, have to change ai code also

File: code/_onclick/click.dm [hunk 2]
```diff
@@ -45,116 +44,110 @@
...
-/mob/proc/ClickOn(atom/A, location, params)
-
+/mob/proc/ClickOn(datum/source, atom/target, location, params)
+	SIGNAL_HANDLER
 	if(world.time <= next_click)
 		return
 	next_click = world.time + 1
 
-	if(check_click_intercept(params, A))
+	if(check_click_intercept(params, target))
```
Comment (BraveMole) [inline code/_onclick/click.dm:53]: probably want to remove all click_intercept while i'm here

File: code/_onclick/click.dm [hunk 3]
```diff
@@ -45,116 +44,110 @@
...
 		return
 
 	if(notransform)
```
Comment (BraveMole) [inline code/_onclick/click.dm:56]: no idea what is this, i'm gonna leave it here

File: code/_onclick/click.dm [hunk 4]
```diff
@@ -45,116 +44,110 @@
...
-		ShiftMiddleClickOn(A)
+		INVOKE_ASYNC(src, .proc/ShiftMiddleClickOn, target)
 		return
 	if(modifiers["shift"] && modifiers["ctrl"])
-		CtrlShiftClickOn(A)
+		INVOKE_ASYNC(src, .proc/CtrlShiftClickOn, target)
 		return
 	if(modifiers["ctrl"] && modifiers["middle"])
-		CtrlMiddleClickOn(A)
+		INVOKE_ASYNC(src, /atom/proc/CtrlMiddleClickOn, target)
```
Comment (BraveMole) [inline code/_onclick/click.dm:73]: mob doesn't have a ctrlmiddleclickon, so i do this. Probably need improvement

File: code/_onclick/click.dm [hunk 5]
```diff
@@ -45,116 +44,110 @@
...
+	if(CanReach(target, held_item))
+		if(held_item)
+			INVOKE_ASYNC(held_item, /obj/item/proc/melee_attack_chain, src, target, params, modifiers["right"])
+			return
+		INVOKE_ASYNC(src, .proc/UnarmedAttack, target, TRUE, modifiers)
+		return
+	
+	if(held_item)
+		var/proximity = target.Adjacent(src)
+		if(!proximity || !target.attackby(held_item, src, params))
```
Comment (BraveMole) [inline code/_onclick/click.dm:143]: This lane has issue, i'm sure some attackby sleep. Probably gonna make it a signal or something

**Comment by InterroLouis** [comment]
Personally I would sometimes mouse down and then actually AVOID firing(while not on automatic) by holding the mouse down and taking it off the left side game screen.

**Comment by BraveMole** [comment]
Later

**Pull request closed without merging**